lib/rules/media-feature-name-value-allowed-list/index.js

```diff
@@ -37,70 +40,98 @@ const rule = (primary) => {
 		}
 
 		root.walkAtRules(/^media$/i, (atRule) => {
-			mediaParser(atRule.params).walk(/^media-feature-expression$/i, (node) => {
-				if (!node.nodes) return;
-
-				const mediaFeatureRangeContext = isRangeContextMediaFeature(node.parent.value);
-
-				// Ignore boolean
-				if (!node.value.includes(':') && !mediaFeatureRangeContext) {
-					return;
-				}
-
-				const mediaFeatureNode = node.nodes.find((n) => n.type === 'media-feature');
-
-				if (mediaFeatureNode == null) throw new Error('A `media-feature` node must be present');
-
-				let mediaFeatureName;
-				let values;
+			parseMediaQuery(atRule).forEach((mediaQuery) => {
+				if (isMediaQueryInvalid(mediaQuery)) return;
+
+				mediaQuery.walk(
```

Comment:
[suggest] This is just my personal preference. You can leave it as-is.

I feel a bit hard to read the code since a deeper indentation and repeated `entry.{var}` variables are present. How about the following refactoring like this?

```js
const initialState = {
	mediaFeatureName: '',
	unprefixedMediaFeatureName: '',
};

mediaQuery.walk(({ node, state }) => {
  // ...
}, initialState);
```

Reply:
Thank you!

I was looking for ways to reduce the indentation a bit, but I didn't consider that the state object at end was causing it :)

Much better with your suggestion 🙇